pyaedt/q3d.py

```diff
@@ -368,6 +370,355 @@ def edit_sources(
 
         return True
 
+    def export_matrix_data(
+        self,
+        file_name,
+        problem_type=None,
+        variations=None,
+        setup_name=None,
+        sweep=None,
+        reduce_matrix=None,
+        r_unit=None,
+        l_unit=None,
+        c_unit=None,
+        g_unit=None,
+        freq=None,
+        freq_unit=None,
+        matrix_type=None,
+        export_AC_DC_res=None,
+        precision=None,
+        field_width=None,
+        use_sci_notation=None,
+        length_settings=None,
+        length=None,
+    ):
+        """Export Matrix Data.
+
+        Parameters
+        ----------
+        file_name : str
+            File path to save matrix data.
+            Possible file extensions are: *.m, *.lvl, *.csv, *.txt.
+        problem_type : str, optional
+            Problem type.
+            Default value is "C".
+            Possible values are: "C", "AC RL", "DC RL".
+        variations : str, optional
+            Design variation.
+            Empty string uses the current nominal variation.
+        setup_name : str, optional
+            Setup Name.
+            Default value is first analysis setup name.
+        sweep : str, optional
+            Solution frequency.
+            Default value is default adaptive.
+        reduce_matrix : str, optional
+            Name of the matrix to display.
+            Default value is "Original".
+        r_unit : str, optional
+            Resistance unit value.
+            Default value is "ohm".
+        l_unit : str, optional
+            Inductance unit value.
+            Default value is "nH".
+        c_unit : str, optional
+            Capacitance unit value.
+            Default value is ``"pF"``.
+        g_unit : str, optional
+            Conductance unit value.
+            Default value is "pF".
+        freq : str, optional
+            Selected frequency.
+            Default Value is 0Hz.
+        freq_unit : str, optional
+            Frequency unit.
+            Default value is the one set in Default unit.
+        matrix_type : str, optional
+            Matrix Type.
+            Possible Values are "Maxwell", "Spice" and "Couple".
+            Default value is "Maxwell, Spice, Couple".
+        export_AC_DC_res : bool, optional
+            Whether to add the AC and DC res.
+            Default value is ``False``.
+        precision : int, optional
+            Precision format.
+            Default value is ``15``.
+        field_width : int, optional
+            Field Width.
+            Default value is 20.
+        use_sci_notation : bool, optional
+            Use sci notation.
+            True to use scientific notation. False to use display format.
+            Default value is True.
+        length_settings : str, optional
+            Length settings if 2D design.
```

Comment:
```suggestion
            Length settings if the design is 2D.
```

Reply:
It seems like it should be length_setting for the parameter name and "Length setting..." in the description. The type is a string--so I assume one value.